cores/esp8266/core_esp8266_i2s.c

```diff
@@ -331,6 +331,45 @@ bool i2s_write_lr(int16_t left, int16_t right){
   return i2s_write_sample(sample);
 }
 
+#define MIN(a,b) (a<b?a:b)
+
+// writes a buffer of samples into the DMA memory, returns the amount of samples written
+int16_t i2s_write_buffer_mono(int16_t *sample, int16_t len) {
```

Comment:
The existing code doesn't do any mono, and has blocking and nonblocking modes.  For a mono array I see why you need a special function.  Can we abstract this, though, to be more in-line with the current options?

i.e.:
````
int16_t _i2s_write_buffer(int16_t *sample, int16_t len, bool mono, bool nb)
{
...existing code, changed to handle mono or stereo in the storing...handle blocking to write all if nb==true...
}

int16_t i2s_write_buffer_mono(int16_t *sample, int16_t len) { return _i2s_write_buffer(sample, len, true, true); }

int16_t i2s_write_buffer_mono_nb(int16_t *sample, int16_t len) { return _i2s_write_buffer(sample, len, true, false); }

int16_t i2s_write_buffer(int16_t *sample, int16_t len) { return _i2s_write_buffer(sample, len, false, true); }

int16_t i2s_write_buffer_nb(int16_t *sample, int16_t len) { return _i2s_write_buffer(sample, len, false false); }
````

Reply:
sounds good, just bear with me because I am still learning how resolving code reviews work on Github :)

Reply:
No problem.  I opened a bug for 2.6.0 to add just this kind of option, assigned to myself since I redid the I2S and maintain a bunch of audio decoders for the 8266.  So you're fixing a bug I assigned to myself, thanks!  Issue #5206 .